### PR TITLE
Don't spread forms. Use hidden inputs instead. 

### LINKED ISSUE #1181: Don't spread forms. Use hidden inputs instead.
https://github.com/taskany-inc/issues/blob/main/src/components/CommentCreateForm/CommentCreateForm.tsx#L65-L67

### PR DESCRIPTION
**Pull request by awinogradov**
## PR includes

- [x] Refactor

## Related issues

Resolve #1181

## QA Instructions, Screenshots, Recordings
  
No degradation

image

### CODE REVIEW

File: src/components/CommentForm/CommentForm.tsx [hunk 1]
```diff
@@ -56,33 +57,55 @@ const StyledCommentForm = styled(FormCard)`
...
-        setCommentFocused(false);
-        onSubmit?.();
-    }, [onSubmit]);
+    }, [onCancel, reset]);
+
+    const onCommentSubmit = useCallback(
+        (form: GoalCommentSchema) => {
+            setBusy(true);
+            setCommentFocused(false);
+            onSubmit?.(form);
```
Comment (9teen90nine) [inline src/components/CommentForm/CommentForm.tsx:104]: try/catch maybe?

File: src/components/CommentEditForm/CommentEditForm.tsx [hunk 2]
```diff
@@ -14,60 +12,36 @@ interface CommentEditFormProps {
...
     const onCommentUpdate = useCallback(
-        (form: CommentUpdate) => {
-            setBusy(true);
-
+        (form: GoalCommentSchema) => {
+            // optimistic update
+            onChange?.({ description: form.description });
             update((comment) => {
                 onUpdate?.(comment);
-                setBusy(false);
```
Comment (9teen90nine) [inline src/components/CommentEditForm/CommentEditForm.tsx:54]: why did you remove this? The form is active during the commit?
there should probably be a try/catch with a fallback also

File: trpc/router/goal.ts [hunk 3]
```diff
@@ -672,6 +672,7 @@ export const goal = router({
 
         if (!commentAuthor) return null;
         if (!actualGoal) return null;
+        if (!input.goalId) return null;
```
Comment (LamaEats) [inline trpc/router/goal.ts:675]: hm, I think this is useless check in this place 'cause `input.goalId` early uses for query from a db 😄 
or you need move this check before query data

File: src/components/CommentForm/CommentForm.tsx (hunk 1 above)
Comment (awinogradov) [reply src/components/CommentForm/CommentForm.tsx:104]: Optimistic update resolves by invalidation goal data

**Review by LamaEats — APPROVED** [review]

File: src/components/CommentEditForm/CommentEditForm.tsx (hunk 2 above)
Comment (awinogradov) [reply src/components/CommentEditForm/CommentEditForm.tsx:54]: We are setting `busy` in CommentForm via onSubmit callback and then `busy` is passed via render props

**Review by 9teen90nine — APPROVED** [review]
